msc_pygeoapi/provider/cangrd_rasterio.py

```diff
@@ -370,6 +370,11 @@ def query(self, range_subset=['TMEAN'], subsets={}, bbox=[],
                                                     datetime_)
                 args['indexes'] = list(range(1, len(date_file_list) + 1))
 
+        if not os.path.isfile(self.data):
```

Comment:
Are we checking whether a path is a file, or whether a file exists on disk?  If the latter:

```python
if not os.path.exists(self.data):
    msg = 'File does not exist'
    LOGGER.error(msg)
    raise ProviderQueryError(msg)
```

Reply:
with `os.path.isfile` we are looking at both, if `self.data` is a file and exists. Isn't more "risky" using only exists, because if `self.data` is a path and exists it will still return True ?

Reply:
It depends on the codepath around `self.data`.  Good point though, let's keep as is.